gym/spaces/box.py

```diff
@@ -28,7 +31,7 @@ def __init__(self, low=None, high=None, shape=None, dtype=None):
                 dtype = np.uint8
             else:
                 dtype = np.float32
-            logger.warn("gym.spaces.Box autodetected dtype as %s. Please provide explicit dtype." % dtype)
+            warnings.warn("gym.spaces.Box autodetected dtype as {}. Please provide explicit dtype.".format(dtype))
```

Comment:
I think it would be better to modify logger.warn method to use warnings

Reply:
You mean replacing the print in https://github.com/openai/gym/blob/master/gym/logger.py#L28 by `warnings.warn` ?
(as you have certainly noticed, I did that edit in order to create a test)